src/v/reflection/adl.h

```diff
@@ -44,11 +46,15 @@ struct adl {
     static constexpr bool is_ss_bool = is_ss_bool_v<T>;
     static constexpr bool is_chrono_milliseconds
       = std::is_same_v<type, std::chrono::milliseconds>;
+    static constexpr bool is_timestamp = std::is_same_v<type, model::timestamp>;
```

Comment:
maybe it would be better to just add `adl` class specialization for both of those types. we do not want this heder to depend on `model` module. should be other way around.
Adding serde as:
```c++
template<>
struct adl<model::timestamp> {
    void to(iobuf& out, model::timestamp&& t);
    model::timestamp from(iobuf_parser& in);
};
```
both specializations can be added to `reflection/adl_serde.h`

Reply:
it's a leftover from the first iteration, removing `model::timestamp` from adl